addons/mail/static/src/js/followers.js

```diff
@@ -239,30 +239,42 @@ var Followers = AbstractField.extend({
             })
             .then(this._reload.bind(this));
     },
+    /**
+     * Remove partners or channels from the followers
+     * @param {Array} [ids.partner_ids] the partner ids
+     * @param {Array} [ids.channel_ids] the channel ids
+     */
+    _messageUnsubscribe: function (ids) {
+        var self = this;
+        var args = [
+            [self.res_id],
+            ids.partner_ids,
+            ids.channel_ids,
+        ];
+        return this._rpc({
+            model: self.model,
+            method: 'message_unsubscribe',
+            args: args
+        }).then(self._reload.bind(self));
+    },
     /**
      * Remove partners or channels from the followers
      * @param {Array} [ids.partner_ids] the partner ids
      * @param {Array} [ids.channel_ids] the channel ids
      */
     _unfollow: function (ids) {
         var self = this;
+        // do not prompt confirmation dialog on unsubscribe of current user.
+        if (_.contains(ids.partner_ids, this.partnerID)) {
```

Comment:
here, you probably want to make the condition a little more precise, like:
```suggestion
        if (_.isEqual(ids.partner_ids, [this.partnerID]) && _.isEmpty(ids.channel_ids)) {
```